pennylane/io/qasm_interpreter.py

```diff
@@ -143,6 +143,13 @@ def generic_visit(self, node: QASMNode, context: dict):
             dict: The context updated after the compilation of all nodes by the visitor.
         """
 
+        # init context
+        context["wires"] = []
+        context["vars"] = dict()
+        context["gates"] = []
+        context["callable"] = None
```

Comment:
Why not
```suggestion
        # init context
        context = {
            "wires": [],
            "vars": {},
            "gates": [],
            "callable": None
        }
```
?

Reply:
It already has a 'name' key prior to this, so I need to add that in, but otherwise I don't see any issue with using whatever syntax you prefer here.

Reply:
In that case, this might be cleaner?
```suggestion
        context.update({
            "wires": [],
            "vars": {},
            "gates": [],
            "callable": None
        })
```

Reply:
It's purely stylistic, so up to you.